File: AlgorithmStudy/src/y2022/oct/Solution1023.java
package y2022.oct;

// https://leetcode.com/problems/set-mismatch/
public class Solution1023 {
    public int[] findErrorNums(int[] nums) {
        int dup = -1;
        int loss = -1;

        for (int i = 0; i < nums.length; i++) {
            if (nums[Math.abs(nums[i]) - 1] < 0) {
                dup = Math.abs(nums[i]);
            } else {
                nums[Math.abs(nums[i]) - 1] *= -1;
            }
        }

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                loss = i + 1;
            }
        }

        return new int[]{dup, loss};
    }
}

File: AlgorithmStudy/src/y2023/oct/Solution1023.kt
package y2023.oct

// https://leetcode.com/problems/power-of-four/?envType=daily-question&envId=2023-10-23
class Solution1023 {
    fun isPowerOfFour(n: Int): Boolean {
        if (n < 1) {
            return false
        }

        val powerOfFourSet = mutableSetOf<Long>()
        var pow = 0L
        var i = 0.0
        while (pow < n) {
            pow = Math.pow(4.0, i).toLong()
            powerOfFourSet.add(pow)
            i++
        }

        return powerOfFourSet.contains(n.toLong())

    }
}